routes/auth.go

```diff
@@ -40,3 +93,287 @@ func AuthMiddleware() gin.HandlerFunc {
+	if err != nil {
+		log.Error().Err(err).Str("refresh_claim", util.Formatted(refreshTokenClaim)).Msg("failed to generate access token")
+		SendError(ctx, http.StatusInternalServerError, errors.New("error generating token"))
+		return
+	}
+
+	log.Info().
+		Str("user_id", accessTokenClaim.User.ID.String()).
+		Str("refresh_token_id", refreshTokenClaim.ID.String()).
+		Str("access_token_id", accessTokenClaim.ID.String()).
+		Str("exp", accessTokenClaim.Exp.String()).
+		Msg("issued new access token")
+
+	response := LoginResponse{
+		User:         userResp,
+		Organization: org,
+		RefreshToken: tokenResp{
+			Token: refreshTokenString,
+			Exp:   refreshTokenClaim.Exp,
+		},
+		AccessToken: tokenResp{
+			Token: accessTokenString,
+			Exp:   accessTokenClaim.Exp,
+		},
+	}
+
+	ctx.JSON(http.StatusOK, response)
+}
+
+type tokenUserOrOrg struct {
+	ID   uuid.UUID `json:"id"`
+	Role string    `json:"role"`
+}
+
+type tokenClaim struct {
+	Exp          time.Time      `json:"exp"`
+	ID           uuid.UUID      `json:"id"`
+	User         tokenUserOrOrg `json:"user"`
+	Organization tokenUserOrOrg `json:"organization"`
+}
+
+func (claim tokenClaim) Valid() error {
+	if claim.Exp.Before(time.Now()) {
+		return jwt.ErrTokenExpired
+	}
+	return nil
+}
+
+type refreshTokenClaim struct {
+	Type string `json:"type"`
+	tokenClaim
+}
+
+func (claim refreshTokenClaim) Valid() error {
+	if claim.Type != "refresh" {
+		return errors.New("not a refresh token")
+	}
+	return claim.tokenClaim.Valid()
+}
+
+type accessTokenClaim struct {
+	Type string `json:"type"`
+	tokenClaim
+}
+
+func (claim accessTokenClaim) Valid() error {
+	if claim.Type != "access" {
+		return errors.New("not an access token")
+	}
+	return claim.tokenClaim.Valid()
+}
+
+func issueRefreshToken(user *userResp, org *organizationResp) (string, refreshTokenClaim, error) {
+	id := uuid.New()
+	exp := time.Now().Add(refreshTokenDuration)
+
+	claims := refreshTokenClaim{
+		Type: "refresh",
+		tokenClaim: tokenClaim{
+			Exp: exp,
+			ID:  id,
+			User: tokenUserOrOrg{
+				ID:   user.ID,
+				Role: user.Role,
+			},
+			Organization: tokenUserOrOrg{
+				ID:   org.ID,
+				Role: org.Role,
+			},
+		},
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, claims)
+	ss, err := token.SignedString(tokenPrivateKey)
+	if err != nil {
+		return "", claims, err
+	}
+
+	return ss, claims, nil
+}
+
+// validateRefreshToken
+func _(tokenString string) (*refreshTokenClaim, error) {
+	token, err := jwt.ParseWithClaims(tokenString, &refreshTokenClaim{}, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodECDSA); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+		return tokenPublicKey, nil
+	})
+
+	if claims, ok := token.Claims.(*refreshTokenClaim); ok && token.Valid {
+		return claims, nil
+	} else {
+		return claims, err
+	}
+}
+
+func issueAccessToken(claim *refreshTokenClaim) (string, accessTokenClaim, error) {
```

Comment:
I don't like the idea to pass here the Refresh-Token, we should not depend on the Refresh-Token to create the Access-Token internally. Yes, on the http api level you can only issue a new Access-Token with the Refresh-Token, but for our internal use that's not needed. What do you think?

Can we use the same function parameters as `func issueRefreshToken(user *userResp, org *organizationResp) ...`?